src/features/utils/feature-base.js

```diff
@@ -16,5 +17,7 @@ export class FeatureBase {
      * @type {boolean}
      */
     this.blocked = false
+
+    this.timeKeeper = TimeKeeper.getTimeKeeperByAgentIdentifier(this.agentIdentifier)
```

Comment:
Just to clarify -- It looks like the plan for the future will be for features to access the timekeeper from their class instance props?

Reply:
Correct. Putting the timekeeper here reduces the amount of code in each feature to go and grab the agent timekeeper and makes it easier to use the timekeeper.